portfolio/src/main/java/com/google/sps/data/BookResponseParser.java

```diff
@@ -0,0 +1,165 @@
+  private static final String INDUSTRTRY_IDS = "industryIdentifiers";
+
+  public static Book parseBook(String jsonResponse) {
+    JSONObject jsonObject = new JSONObject(jsonResponse);
+    return jsonToBook(jsonObject);
+  }
+
+  private static Book jsonToBook(JSONObject jsonObject) throws JSONException {
+    Book.Builder builder = Book.builder();
+    JSONObject volumeInfo = jsonObject.getJSONObject(VOLUME_INFO);
+
+    try {
+      String isbn = getIsbn(volumeInfo).get(ISBN);// double nested [{}]
+      System.out.println(isbn);
+      if (isbn == null) {
+        //should never be the case, all books should have ISBN-13
+        return null;
+      }
+      builder.isbn(isbn);
+
+      if (volumeInfo.has(TITLE)) {
+        String title = volumeInfo.get(TITLE).toString();
+        builder.title(title);
+      }
+
+      if (volumeInfo.has(LANGUAGE)) {
+        String lang = volumeInfo.get(LANGUAGE).toString();
+        builder.language(lang);
+      }
+
+      if (volumeInfo.has(DESCRIPTION)) {
+        String desc = volumeInfo.get(DESCRIPTION).toString();
+        builder.description(desc);
+      }
+
+      if (volumeInfo.has(INFO_LINK)) {
+        String infoLink = volumeInfo.get(INFO_LINK).toString();
+        builder.infoLink(infoLink);
+      }
+
+      if (volumeInfo.has(PAGE_COUNT)) {
+        int pageCount =
+            Integer.parseInt(volumeInfo.get(PAGE_COUNT).toString());
+        builder.pageCount(pageCount);
+      }
+
+      if (volumeInfo.has(PUBLISHED_DATE)) {
+        String pubDate = volumeInfo.get(PUBLISHED_DATE).toString();
+        builder.publishedDate(pubDate);
+      }
+
+      if (volumeInfo.has(PUBLISHER)) {
+        String publisher = volumeInfo.get(PUBLISHER).toString();
+        builder.publisher(publisher);
+      }
+
+      if (volumeInfo.has(MATURITY_RATING)) {
+        String matRate = volumeInfo.get(MATURITY_RATING).toString();
+        builder.maturityRating(matRate);
+      }
+
+      if (volumeInfo.has(CATEGORIES)) {
+        ArrayList<String> categories =
+            jsonArrayToStringArray(volumeInfo.getJSONArray(CATEGORIES));
+        builder.categories(categories);
+      }
+
+      if (volumeInfo.has(AUTHORS)) {
+        ArrayList<String> authors =
+            jsonArrayToStringArray(volumeInfo.getJSONArray(AUTHORS));
+        builder.authors(authors);
+      }
+
+      // nested in map {}
+      if (volumeInfo.has(IMAGE_LINKS)
+          && volumeInfo
+                 .getJSONObject(IMAGE_LINKS)
+                 .has(THUMBNAIL)) { // if imageLinks and the thumbnail are present
+        String thumbnail = volumeInfo
+                            .getJSONObject(IMAGE_LINKS)
+                            .get(THUMBNAIL)
+                            .toString();
+        builder.thumbnail(thumbnail);
+      }
+    } catch (JSONException e) {
+        e.printStackTrace();
+        throw new JSONException("There was an error when building the book from the json string");
```

Comment:
You can remove all references of `e.printStackTrace()`, as this will not be useful when deployed on server, and change the throw to include the original exception:

`throw new JSONException("There was an error when building the book from the json string", e);`